commands/server/start.go

```diff
@@ -72,17 +72,28 @@ func StartCmd(gen AppGenerator, logger log.Logger, home string, args []string) e
 	if err != nil {
 		return errors.Wrap(err, "failed to create a listener")
 	}
+
 	svr.SetLogger(logger.With("module", "abci-server"))
-	err = svr.Start()
-	if err != nil {
-		return errors.Wrap(err, "failed to start a server")
-	}
+
 	done := make(chan bool)
-	cmn.TrapSignal(logger, func() {
+	cleanupCallback := func() {
 		// Cleanup
 		_ = svr.Stop()
 		done <- true
-	})
+	}
+
+	cmn.TrapSignal(logger, cleanupCallback)
+
+	defer func() {
+		if r := recover(); r != nil {
+			cleanupCallback()
```

Comment:
req: it is good to call cleanup here but I think we will loose the stack traces of the original panic as `r` is not further processed. Please log them.

Reply:
good call! done